src/components/com_tjcertificate/site/controllers/externalcertificate.php

```diff
@@ -0,0 +1,264 @@
+		$form = $model->getForm($data, false);
+
+		if (!$form)
+		{
+			throw new \Exception($model->getError(), 500);
+		}
+
+		$validData = $model->validate($form, $data);
+
+		// Check for validation errors.
+		if ($validData === false)
+		{
+			// Get the validation messages.
+			$errors = $model->getErrors();
+
+			if (!empty($errors))
+			{
+				// Push up to three validation messages out to the user.
+				for ($i = 0, $n = count($errors); $i < $n && $i < 3; $i++)
+				{
+					if ($errors[$i] instanceof Exception)
+					{
+						$app->enqueueMessage($errors[$i]->getMessage(), 'warning');
+					}
+					else
+					{
+						$app->enqueueMessage($errors[$i], 'warning');
+					}
+				}
+			}
+
+			// Save the data in the session.
+			$app->setUserState('com_tjcertificate.edit.externalcertificate.data', $data);
+
+			// Redirect back to the edit screen.
+			$redirectUrl = Route::_('index.php?option=com_tjcertificate&view=externalcertificate' . $this->getRedirectToItemAppend($recordId, $urlVar), false);
+			$this->setRedirect($redirectUrl);
+
+			return false;
+		}
+
+		if ($validData['assigned_user_id'])
+		{
+			$validData['user_id'] = $validData['assigned_user_id'];
+		}
+		else
+		{
+			$validData['user_id'] = $user->id;
+		}
+
+		$validData['client'] = "external";
+		$validData['state'] = 0;
+		$validData['is_external'] = 1;
+
+		$file = $app->input->files->get('jform', array(), 'array');
+
+		if (!empty($file['cert_file']))
+		{
+			$validData['old_media_ids'] = $app->input->get('oldFiles', 0, 'INT');
+			$uploadData = $model->uploadMedia($file, $validData);
+			$validData['cert_file'] = $uploadData['source'];
+		}
+
+		$certificateModel = TJCERT::model('Certificate', array('ignore_request' => true));
+
+		$certificateModel->save($validData);
+
+		$modelMediaXref = TJMediaXref::getInstance();
+
+		if ($uploadData['id'])
+		{
+			$mediaData['id'] = '';
+			$mediaData['client_id'] = $certificateModel->getState('certificate.id');
+			$mediaData['media_id'] = $uploadData['id'];
+			$mediaData['client'] = 'com_tjcertificate';
+			$modelMediaXref->bind($mediaData);
+			$modelMediaXref->save();
+		}
+
+		$this->setRedirect(JRoute::_('index.php?option=com_tjcertificate&view=certificates&layout=my', false));
+	}
+
+	/**
+	 * Cancel operation
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function cancel()
+	{
+		// Check for request forgeries.
+		$this->checkToken('request');
+		$this->setRedirect(JRoute::_('index.php?option=com_tjcertificate&view=certificates&layout=my', false));
+	}
+
+	/**
+	 * Method to publish a list of articles.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function publish()
+	{
+		parent::publish();
+
+		$this->setRedirect('index.php?option=com_tjcertificate&view=certificates&layout=my');
+	}
+
+	/**
+	 * Method to download certificate.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function downloadCertificate()
```

Comment:
we already have a function to download certificte

Reply:
there is for pdf download